backend/src/test/java/de/ffmjava/capstone/backend/stock/StockServiceTest.java

```diff
@@ -73,35 +79,63 @@ void addNewStockItem_AndExpectStockItem_200() {
+        when(mockStockRepository.existsByName("name")).thenReturn(true);
 
         assertThrows(StockItemAlreadyExistsException.class, () -> service.addNewStockItem(newStockItem));
     }
 
     @Test
     void deleteStockItem_AndExpectSuccess() {
         String idToDelete = "1";
-        when(mockRepository.existsById(idToDelete)).thenReturn(true);
-        doNothing().when(mockRepository).deleteById(idToDelete);
+        when(mockStockRepository.existsById(idToDelete)).thenReturn(true);
+        doNothing().when(mockStockRepository).deleteById(idToDelete);
 
         assertTrue(service.deleteStockItem(idToDelete));
-        verify(mockRepository).existsById(idToDelete);
+        verify(mockStockRepository).existsById(idToDelete);
     }
 
     @Test
     void deleteStockItem_AndExpectException_404() {
         String idToDelete = "1";
         ResponseStatusException expectedException = new ResponseStatusException(HttpStatus.NOT_FOUND, "Kein Eintrag für die gegebene ID gefunden");
-        when(mockRepository.existsById(idToDelete))
+        when(mockStockRepository.existsById(idToDelete))
                 .thenReturn(false);
-        doNothing().when(mockRepository).deleteById(idToDelete);
+        doNothing().when(mockStockRepository).deleteById(idToDelete);
         try {
             service.deleteStockItem(idToDelete);
             fail();
         } catch (ResponseStatusException e) {
             assertEquals(expectedException.getMessage(), e.getMessage());
-            verify(mockRepository).existsById(idToDelete);
+            verify(mockStockRepository).existsById(idToDelete);
         }
     }
 
+    @Test
+    void deleteStockItem_noConsumptionToDelete_200() {
+        String idToDelete = "1";
+        when(mockStockRepository.existsById(idToDelete))
+                .thenReturn(true);
+        when(mockHorseRepository.findHorsesByConsumptionId("1")).thenReturn(new ArrayList<>());
+        assertTrue(service.deleteStockItem("1"));
+    }
+
+    @Test
+    void deleteStockItem_cascadingDeleteConsumption_200() {
+        String idToDelete = "1";
+        Horse horseWithoutConsumption = new Horse("id", "name", "owner", new ArrayList<>());
+        Horse horseWithConsumption = horseWithoutConsumption
+                .withConsumptionList(new ArrayList<>(List.of(new Consumption("1", "name", new BigDecimal("0")))));
+        List<Horse> retrievedHorses = new ArrayList<>(List.of(horseWithConsumption));
+
+        when(mockStockRepository.existsById(idToDelete))
+                .thenReturn(true);
+        when(mockHorseRepository.findHorsesByConsumptionId(idToDelete))
+                .thenReturn(retrievedHorses);
+        when(mockHorseRepository.saveAll(List.of(horseWithoutConsumption)))
+                .thenReturn(List.of(horseWithoutConsumption));
+
+        assertTrue(service.deleteStockItem("1"));
+        verify(mockHorseRepository).findHorsesByConsumptionId(any());
```

Comment:
also without given/when/then harder to read. Again overlooked where the service gets called